rotkehlchen/chain/ethereum/modules/liquity/trove.py

```diff
@@ -84,8 +84,32 @@ def __init__(
     def get_positions(
             self,
             given_addresses: Sequence[ChecksumEvmAddress],
-    ) -> dict[ChecksumEvmAddress, Trove]:
-        """Query liquity contract to detect open troves"""
+    ) -> dict[str, Any]:
+        """Query liquity contract to get the collateral ratio and detect open troves"""
+
+        eth_price = Inquirer().find_usd_price(A_ETH)
+        lusd_price = Inquirer().find_usd_price(A_LUSD)
+
+        try:
+            system_collateral = self.trove_manager_contract.call(
+                node_inquirer=self.ethereum,
+                method_name='getEntireSystemColl',
+                arguments=[],
+            )
+            system_debt = self.trove_manager_contract.call(
+                node_inquirer=self.ethereum,
+                method_name='getEntireSystemDebt',
+                arguments=[],
+            )
+
+            if system_debt != 0:
+                final_collateral_ratio = eth_price * system_collateral / system_debt * lusd_price
+            else:
+                final_collateral_ratio = None
+
+        except RemoteError as e:
+            log.debug(f'Failed {e!s}')
```

Comment:
could you move this to a "private" method to keep this logic cleaner?

Reply:
Also when moving it make it take the price as argument and restore the call to where it was. I believe the reason is that if any remote query fails you don't make extra price queries that won't be used